solver/src/solver/paraswap_solver.rs

```diff
@@ -57,40 +60,132 @@ impl ParaswapSolver {
     }
 }
 
+fn is_retryable_err(err: &SettlementError) -> bool {
+    match err {
+        SettlementError::Paraswap(ParaswapResponseError::PriceChange) => {
+            tracing::warn!("Paraswap Price Change");
+            true
+        }
+        SettlementError::Paraswap(ParaswapResponseError::BuildingTransaction(query)) => {
+            tracing::warn!("Paraswap ERROR_BUILDING_TRANSACTION from query {}", query);
+            true
+        }
+        _ => false,
+    }
+}
+
+enum SettlementError {
```

Comment:
naming nit (since this is internal): To me this enum is all kinds of `ParaswapError`, with subtypes as of right now  being `TransactionCreation(ParaswapResponseError)` or `Other(anyhow::Error)`.

An alternative way (which could still be coerced from the two subtype errors) of representing this data structure would be, but the enum is totally fine:

```rust
struct ParaswapError {
  inner: anyhow::Error,
  is_retryable: bool,
}
```

Reply:
I very much like this idea although it looks like the inner would have to be the enum we have recently introduced since anyhow doesn't include `ParaswapResponseError`. Am I missing something here? Shouldn't I be putting the `is_retryable`also on the `ParaswapResponseError`?